src/module/CandidateSplitting.cpp

```diff
@@ -0,0 +1,154 @@
+#include "crpropa/module/CandidateSplitting.h"
+
+namespace crpropa {
+
+CandidateSplitting::CandidateSplitting() {
+	// no particle splitting if EnergyBins and NSplit are not specified
+	setNsplit(0);
+	setMinimalWeight(1.);
+}
+
+CandidateSplitting::CandidateSplitting(int n_split, double Emin, double Emax, double n_bins, double minWeight) {
+	// linear energy bins:
+	setNsplit(n_split);
+	setEnergyBins(Emin, Emax, n_bins, false);
+	setMinimalWeight(minWeight);
+}
+
+CandidateSplitting::CandidateSplitting(int n_split, double Emin, double Emax,  double n_bins, double minWeight, bool log) {
+	setNsplit(n_split);
+	setEnergyBins(Emin, Emax, n_bins, log);
+	setMinimalWeight(minWeight);
+}
+
+CandidateSplitting::CandidateSplitting(double SpectralIndex, double Emin, int nBins)  {
+	// to use with Diffusive Shock Acceleration
+
+	if (SpectralIndex <= 0){
```

Comment:
What spectral index is meant? The the exponent or just the positive value. 
In other modules like the `SourcePowerLawIndex` we assume the index for $E^{-1}$ to be `-1`.

Reply:
Here it's the absolute value, as stated in CandidateSplitting.h

Reply:
But I changed it to be negative now